src/client.rs

```diff
@@ -28,16 +28,37 @@ pub struct Client {
 }
 
 impl Client {
-    pub async fn new() -> Result<(Self, RunParameters), Box<dyn std::error::Error>> {
+    pub async fn new_and_init() -> Result<(Self, RunParameters), Box<dyn std::error::Error>> {
         let params = RunParameters::try_parse()?;
 
         let (cmd_tx, cmd_rx) = channel(1);
 
         let background = BackgroundTask::new(cmd_rx, params.clone()).await?;
+        let client = Self { cmd_tx };
 
         tokio::spawn(background.run());
 
-        Ok((Self { cmd_tx }, params))
+        client.wait_network_initialized().await?;
+
+        let global_seq_num = client
+            // Note that the sdk-go only signals, but not waits.
+            .signal_and_wait("initialized_global", params.test_instance_count)
+            .await?;
+
+        let group_seq_num = client
+            // Note that the sdk-go only signals, but not waits.
+            .signal_and_wait(
+                format!("initialized_group_{}", params.test_group_id),
+                params.test_group_instance_count as u64,
+            )
+            .await?;
+
+        client.record_message(format!(
+            "claimed sequence numbers; global={}, group({})={}",
+            global_seq_num, params.test_group_id, group_seq_num
+        ));
+
+        Ok((client, params))
     }
```

Comment:
Looks good, thanks for sharing the draft!

A couple of notes:

- the go sdk doesn't wait in the init, but provides [methods to do so](https://github.com/testground/sdk-go/blob/49c90fa754052018b70c63d87b7f1d37f6080a78/run/init_context.go#L73-L98)
  - I can't see a good reason to need this level of granularity, I guess we'll port this behavior to the go-sdk
- the go sdk sends the global seq and group seq to the test
  - It might be useful to implement leader/follower quickly. [Related discussion](https://github.com/testground/testground/issues/1336)
  - I've seen it used in another test as [seed value](https://github.com/libp2p/test-plans/blob/master/ping/main.go#L251-L253), but that use case seems anecdotal

Reply:
> the go sdk sends the global seq and group seq to the test

:+1: for exposing it to the test, though I haven't clean way to do so yet. Still want to put more thoughts into it. In my eyes this can happen as a follow up.

Reply:
> * I can't see a good reason to need this level of granularity, I guess we'll port this behavior to the go-sdk

Same here. I can't think of a use-case where one would want to publish but not wait.